IdentityCore/src/util/mac/MSIDXpcSingleSignOnProvider.m

```diff
@@ -394,14 +394,27 @@ - (void)getXpcService:(id<MSIDXpcProviderCaching>)xpcProviderCache withContinueB
         return;
     }
     
+    // Ensure that both the interruption handler and invalidation handler do not trigger unexpected dispatch_group_leave
+    // when the connection is unavailable or rejected by the XPC. This is achieved by adding a manual check.
+    __block BOOL isConnectionErroredOut = NO;
```

Comment:
Is this change to make sure when InterruptionHandler and InvalidationHandler are both called, the continueBlock is only called once? how often does it happen?